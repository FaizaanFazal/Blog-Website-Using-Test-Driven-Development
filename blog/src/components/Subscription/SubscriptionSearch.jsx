import React, { useState } from 'react'

export default function SubscriptionSearch() {
    const [emailValue, setEmailValue] = useState('');

    const handleEmailInput = (e) => {
        e.preventDefault();
        setEmailValue(e.target.value);
      };

    return (
        <form data-testid="form" className="newsletter-form">
            <div className="form-group flex items-stretch">
                <input 
                    type="email"
                    data-testid="email"
                    value={emailValue}
                    className="form-input"
                    placeholder="Your Email" 
                    onChange={handleEmailInput}
                />
                <button data-testid="subscribebtn" type="submit" className="btn btn-primary nowrap">Get Started</button>
            </div>
            <p className="text text-base text-gray">
                Get a response tomorrow if you submit by 9pm today. If we received after 9pm will get a response the following data.
            </p>
        </form>
    )
}
