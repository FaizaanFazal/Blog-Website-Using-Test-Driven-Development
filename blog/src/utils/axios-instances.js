// import axios from 'axios';

// export const axiosInstance = axios.create({
//   baseURL: 'http://localhost:8000/',
// });
import axios from 'axios';

const placeholderApi = axios.create({
  baseURL: 'http://localhost:8000',
});

export default placeholderApi;
