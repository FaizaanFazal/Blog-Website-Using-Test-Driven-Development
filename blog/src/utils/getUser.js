export const getUser = () => {
    const userString = localStorage.getItem('user');
    if (userString) {
      const user = JSON.parse(userString);
      return user;
    } else {
      return null;
    }
  };