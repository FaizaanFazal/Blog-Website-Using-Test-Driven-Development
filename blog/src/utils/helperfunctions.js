import { fireEvent } from '@testing-library/react';
import images from './images';

export const isProperImageURL = (url) => {
  /* eslint-disable-next-line no-useless-escape */
  const regex = /^(https?|ftp):\/\/(([a-z\d]([a-z\d-]*[a-z\d])?\.)+[a-z]{2,}|localhost)(\/[-a-z\d%_.~+]*)*(\?[;&a-z\d%_.~+=-]*)?(\#[-a-z\d_]*)?$/i;
  return regex.test(url);
};
export const isProperEmail = (email) => {
  const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return emailRegex.test(email);
};
export const isProperPass = (pass) => {
  const passregex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/;
  return passregex.test(pass);
};
export function backspace(element) {
  let actuallyTyped = element.value;

  const backspaceKey = {
    key: 'Backspace',
    code: 8,
    inputType: 'deleteContentBackward',
  };

  const sharedEventConfig = {
    key: backspaceKey.key,
    charCode: backspaceKey.code,
    keyCode: backspaceKey.code,
    which: backspaceKey.code,
    modifier: backspaceKey.modifier,
  };
  const downEvent = fireEvent.keyDown(element, sharedEventConfig);

  if (downEvent) {
    actuallyTyped = actuallyTyped.slice(0, -1);

    fireEvent.input(element, {
      target: { value: actuallyTyped },
      inputType: backspaceKey.inputType,
      bubbles: true,
      cancelable: true,
    });
  }

  fireEvent.keyUp(element, sharedEventConfig);
}

export const small = [
  {
    id: 1,
    title: '10 Tips for Budget-Friendly Travel',
    author: 'Elena Rodriguez',
    authorImage: { src: images.author1, alt: 'Elena Rodriguez' },
    date: '15 August',
    image: { src: images.blogImg1, alt: '10 Tips for Budget-Friendly Travel Blog Header Image' },
    verified: true,
    slug: 'post1',
    content: "test Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.",
  },
];
export const greater = [
  {
    id: 1,
    title: '10 Tips for Budget-Friendly Travel',
    author: 'Elena Rodriguez',
    authorImage: { src: images.author1, alt: ' Elena' },
    date: '15 August',
    image: { src: images.blogImg1, alt: '10 Tips for Budget-Friendly Travel Blog Header Image' },
    slug: 'post1',
    verified: true,
    content: "Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank. \n \n Traveling on a budget doesnt have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank. \n \n Traveling on a budget doesnt have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank. \n",
  }];
