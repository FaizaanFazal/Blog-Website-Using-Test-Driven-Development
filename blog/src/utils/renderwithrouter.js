import {render} from '@testing-library/react'
import * as userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'

export const renderWithRouter = (ui, {route = '/'} = {}) => {
  window.history.pushState({}, 'Test page', route)

  return {
    user: userEvent.setup,
    ...render(ui, {wrapper: MemoryRouter}),
  }
}