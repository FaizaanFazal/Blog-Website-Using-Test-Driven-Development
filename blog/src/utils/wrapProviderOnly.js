import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import reduxStore from '../Redux/store';
import '@testing-library/jest-dom';

export * from '@testing-library/react';

const AllTheProviders = ({ children }) => (
  <Provider store={reduxStore}>
    {children}
  </Provider>
);

export const renderWithProvidersOnly = (ui, options) => render(<AllTheProviders>{ui}</AllTheProviders>, options);

// export function renderWithProviders(
//   ui,
//   {
//     // preloadedState = {},
//     // Automatically create a store instance if no store was passed in
//     store = reduxStore,
//     ...renderOptions
//   } = {},
// ) {
//   function Wrapper({ children }) {
//     return <Provider store={store}>{children}</Provider>;
//   }

//   // Return an object with the store and all of RTL's query functions
//   return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
// }

renderWithProvidersOnly.propTypes = {
  children: PropTypes.node.isRequired,
};
