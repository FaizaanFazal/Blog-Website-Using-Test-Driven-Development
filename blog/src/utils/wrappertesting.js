import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import reduxStore from '../Redux/store';
import '@testing-library/jest-dom';

export * from '@testing-library/react';

export function renderWithProviders(
  ui,
  {
    // preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = reduxStore,
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

renderWithProviders.propTypes = {
  children: PropTypes.node.isRequired,
};
