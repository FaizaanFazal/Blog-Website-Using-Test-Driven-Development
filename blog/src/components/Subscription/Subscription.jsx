import React from 'react';
import SubscriptionSearch from './SubscriptionSearch';
import images from '../../utils/images';

export default function Subscription() {
  return (
    <>
      <div className="newsletter-curve">
        <img src={`${images.newsletter_wave}`} />
      </div>

      <section className="newsletter-sc bg-blue-dark text-white">
        <div className="container">
          <div className="newsletter-content grid-cols grid-cols-2 items-center">
            <div className="newsletter-l">
              <h3 className="title title-lg">
                Get our stories delivered from us to your inbox weekly.
              </h3>
              <SubscriptionSearch />
            </div>

            <div className="newsletter-r">
              <div className="newsletter-info">
                <div className="newsletter-info-wrapper bg-white">
                  <div className="info-img">
                    <img data-testid="Image" src = { `${images.newsletter}`} className="object-fit-cover" />
                  </div>
                  <div className="info-text">
                    <h4 className="text-xxl text title">
                      The best articles every week
                    </h4>
                    <p className="text text-lg">Our insurnace plans offers are priced the same everywheres else.</p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </>
  );
}
