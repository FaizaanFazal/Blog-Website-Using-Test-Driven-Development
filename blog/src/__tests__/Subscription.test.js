import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { renderWithProviders } from '../utils/wrappertesting';
import Subscription from '../components/Subscription/Subscription';

describe('Unit tests for Subscription Component', () => {

    it('snapshot test to check it renders Subscription component', () => {
        const { asFragment } = render(<MemoryRouter><Subscription /></MemoryRouter>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Image renders in Subscription component', () => {
        renderWithProviders(<MemoryRouter><Subscription /></MemoryRouter>);
        const image = screen.getByTestId('Image');
        expect(image).toBeInTheDocument();
    });
    
    it('Form renders in component', () => {
        renderWithProviders(<MemoryRouter><Subscription /></MemoryRouter>);
        const formelement = screen.getByTestId('form');
        expect(formelement).toBeInTheDocument();  
    });

    it('Email input changes on input', () => {
        renderWithProviders(<MemoryRouter><Subscription /></MemoryRouter>);
        const email = screen.getByTestId('email');
        fireEvent.change(email, { target: { value: 'test' } });
        expect(email.value).toBe('test');
    });
    it('Button renders in subscitpion component', () => {
        renderWithProviders(<MemoryRouter><Subscription /></MemoryRouter>);
        const subscribebtn = screen.getByTestId('subscribebtn');
        expect(subscribebtn).toBeInTheDocument();  
    });

});

