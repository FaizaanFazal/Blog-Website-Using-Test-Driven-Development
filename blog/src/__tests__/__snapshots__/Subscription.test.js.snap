// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Subscription Component snapshot test to check it renders Subscription component 1`] = `
<DocumentFragment>
  <div
    class="newsletter-curve"
  >
    <img
      src="newsletter_wave.svg"
    />
  </div>
  <section
    class="newsletter-sc bg-blue-dark text-white"
  >
    <div
      class="container"
    >
      <div
        class="newsletter-content grid-cols grid-cols-2 items-center"
      >
        <div
          class="newsletter-l"
        >
          <h3
            class="title title-lg"
          >
            Get our stories delivered from us to your inbox weekly.
          </h3>
          <form
            class="newsletter-form"
            data-testid="form"
          >
            <div
              class="form-group flex items-stretch"
            >
              <input
                class="form-input"
                data-testid="email"
                placeholder="Your Email"
                type="email"
                value=""
              />
              <button
                class="btn btn-primary nowrap"
                data-testid="subscribebtn"
                type="submit"
              >
                Get Started
              </button>
            </div>
            <p
              class="text text-base text-gray"
            >
              Get a response tomorrow if you submit by 9pm today. If we received after 9pm will get a response the following data.
            </p>
          </form>
        </div>
        <div
          class="newsletter-r"
        >
          <div
            class="newsletter-info"
          >
            <div
              class="newsletter-info-wrapper bg-white"
            >
              <div
                class="info-img"
              >
                <img
                  class="object-fit-cover"
                  data-testid="Image"
                  src="newsletter.png"
                />
              </div>
              <div
                class="info-text"
              >
                <h4
                  class="text-xxl text title"
                >
                  The best articles every week
                </h4>
                <p
                  class="text text-lg"
                >
                  Our insurnace plans offers are priced the same everywheres else.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
